tests/test_benchcab.py

```diff
@@ -0,0 +1,61 @@
+"""`pytest` tests for benchcab.py"""
+
+import contextlib
+import io
+from subprocess import CalledProcessError
+import pytest
+
+from benchcab.benchcab import Benchcab
+from benchcab import internal
+from benchcab.utils.subprocess import SubprocessWrapperInterface
+from .common import MockSubprocessWrapper, get_mock_config, MOCK_CWD
+
+
+def get_mock_app(
+    subprocess_handler: SubprocessWrapperInterface = MockSubprocessWrapper(),
+) -> Benchcab:
+    """Returns a mock `Benchcab` instance for testing against."""
+    config = get_mock_config()
+    app = Benchcab(argv=["benchcab", "fluxnet"], config=config, validate_env=False)
+    app.subprocess_handler = subprocess_handler
+    app.root_dir = MOCK_CWD
+    return app
+
+
+def test_fluxnet_submit_job():
+    """Tests for `Benchcab.fluxnet_submit_job()`."""
+
+    # Success case: test qsub command is executed
+    mock_subprocess = MockSubprocessWrapper()
+    app = get_mock_app(mock_subprocess)
+    app.fluxnet_submit_job()
+    assert f"qsub {MOCK_CWD / internal.QSUB_FNAME}" in mock_subprocess.commands
+
+    # Success case: test non-verbose output
+    app = get_mock_app()
+    with contextlib.redirect_stdout(io.StringIO()) as buf:
+        app.fluxnet_submit_job()
+    assert buf.getvalue() == (
+        "Creating PBS job script to run FLUXNET tasks on compute "
+        f"nodes: {internal.QSUB_FNAME}\n"
+        "PBS job submitted: mock standard output\n"
+        "The CABLE log file for each task is written to "
+        f"{internal.SITE_LOG_DIR}/<task_name>_log.txt\n"
+        "The CABLE standard output for each task is written to "
+        f"{internal.SITE_TASKS_DIR}/<task_name>/out.txt\n"
+        "The NetCDF output for each task is written to "
+        f"{internal.SITE_OUTPUT_DIR}/<task_name>_out.nc\n"
+    )
+
+    # Failure case: qsub non-zero exit code prints an error message
+    mock_subprocess = MockSubprocessWrapper()
+    mock_subprocess.error_on_call = True
+    app = get_mock_app(subprocess_handler=mock_subprocess)
+    with contextlib.redirect_stdout(io.StringIO()) as buf:
+        with pytest.raises(CalledProcessError):
+            app.fluxnet_submit_job()
+    assert buf.getvalue() == (
+        "Creating PBS job script to run FLUXNET tasks on compute "
+        f"nodes: {internal.QSUB_FNAME}\n"
+        "Error when submitting job to NCI queue\nmock standard output\n"
```

Comment:
```suggestion
        f"Error when submitting job to NCI queue\n{mock_subprocess.stdout}\n"
```

Reply:
Fixed in [95f1871](https://github.com/CABLE-LSM/benchcab/pull/99/commits/95f187105c25613915dc49b0185b4aebefff0f42)